src/main/java/wooteco/subway/domain/fare/distanceStrategy/DistanceDiscountFactory.java

```diff
@@ -0,0 +1,28 @@
+package wooteco.subway.domain.fare.distanceStrategy;
+
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+public enum DistanceDiscountFactory {
+    BELOW_MINIMUM (discount -> discount < 10, new FreeDiscountPolicy()),
+    BELOW_MAXIMUM (discount -> discount >= 10 && discount < 50, new NormalDistanceDiscountPolicy()),
+    OVER_MAXIMUM (discount -> discount >= 50, new ExtraDiscountPolicy())
+    ;
+
+    private final Predicate<Integer> discountCondition;
+    private final DistanceDiscountPolicy distanceDiscountPolicy;
+
+    DistanceDiscountFactory(Predicate<Integer> discountCondition,
+        DistanceDiscountPolicy distanceDiscountPolicy) {
```

Comment:
```suggestion
    DistanceDiscountFactory(Predicate<Integer> discountCondition, DistanceDiscountPolicy distanceDiscountPolicy) {
```
한 줄로 작성해도 120자가 넘지 않는데 한 줄로 작성해보면 어떨까요?